Added testing to rendered-target

SillyInventor opened the pull request:
Added a fake renderer with known predictable output
Added testing for most if not all functions

### Resolves

Partially addresses #471

### Proposed Changes

Add unit testing to rendered-target

### Reason for Changes

Everybody likes testing

### Test Coverage

Yup.

SillyInventor committed:
Added testing to rendered-target
Added a fake renderer with known predictable output
Added testing for most if not all functions

thisandagain commented:
Sorry for the slow turnaround on this PR @SillyInventor. This looks great!

thisandagain commented on test/unit/sprites_rendered-target.js:
```diff
@@ -1,6 +1,66 @@
+var Runtime = require('../../src/engine/runtime');
+
+
+var fakeRenderer = function (){
```
Inconsistent spacing after `()`.

thisandagain replied:
I feel like the "fakeRenderer" might be better split out and added as a module to "test/fixtures". Thoughts?

thisandagain commented on test/unit/sprites_rendered-target.js:
```diff
@@ -1,6 +1,66 @@
 var Sprite = require('../../src/sprites/sprite');
+var Runtime = require('../../src/engine/runtime');
+
+
```
Extra line break

thisandagain reviewed (changes requested):
This is great. Really minor comments / nitpicks.

SillyInventor committed:
Implemented Proposed Fixes
Moved 'fake renderer' to fixtures
Fixed spacing after ()
Removed extra line break

SillyInventor commented:
It is I who should apologize for the timing of the commit. I've been very busy testing tutorial layouts with students etc. and just sitting on this rather than committing. Hopefully the edits are good.

thisandagain commented:
Looks great @SillyInventor. Thanks for splitting that out. Just for documentation:

## Before
![image](https://cloud.githubusercontent.com/assets/747641/25341456/d7cd1210-28d6-11e7-99ec-9571972d50b5.png)

## After
![image](https://cloud.githubusercontent.com/assets/747641/25341468/e15e817e-28d6-11e7-9f7e-b0d6fab4aa6c.png)
